src/logger/loggerTaskEx.c

```diff
@@ -108,24 +102,28 @@ static void logging_stopped()
 
 void startLoggerTaskEx(int priority)
 {
-    xTaskCreate( loggerTaskEx,( signed portCHAR * ) "logger",	LOGGER_STACK_SIZE, NULL, priority, NULL );
+    xTaskCreate(loggerTaskEx, ( signed portCHAR * ) "logger",
+                 LOGGER_STACK_SIZE, NULL, priority, NULL );
 }
 
-static size_t initSampleRecords(LoggerConfig *loggerConfig)
+static int init_sample_ring_buffer(LoggerConfig *loggerConfig)
 {
-    size_t channelSampleCount = get_enabled_channel_count(loggerConfig);
-
-    for (size_t i=0; i < LOGGER_MESSAGE_BUFFER_SIZE; i++) {
-        LoggerMessage *msg = (g_sampleRecordMsgBuffer + i);
-        msg->type = LoggerMessageType_Sample;
-        if (msg->channelSamples != NULL) {
-            vPortFree(msg->channelSamples);
+        const size_t channel_count = get_enabled_channel_count(loggerConfig);
+        struct sample *s = g_sample_buffer;
+        const struct sample * const end = s + LOGGER_MESSAGE_BUFFER_SIZE;
+        int i;
+
+        for (i = 0; s < end; ++s, ++i) {
+                const size_t bytes = init_sample_buffer(s, channel_count);
+                if (bytes) {
```

Comment:
Yep.  And this is why I often do the `bytes == 0` check as that is harder to mess up